src/Storages/ObjectStorage/DataLakes/Iceberg/ManifestFile.cpp

```diff
@@ -300,7 +325,47 @@ ManifestFileContent::ManifestFileContent(
                     break;
             }
         }
-        this->files.emplace_back(status, added_sequence_number, file, partition_key_value, columns_infos);
+        switch (content_type)
+        {
+            case FileContentType::DATA:
+                this->data_files.emplace_back(
+                    file_path_key,
+                    file_path,
+                    status,
+                    added_sequence_number,
+                    partition_key_value,
+                    common_partition_specification,
+                    columns_infos,
+                    DataFileSpecificInfo{});
+                break;
+            case FileContentType::POSITIONAL_DELETE: {
+                std::optional<String> reference_file_path = std::nullopt;
+                try
+                {
+                    // reference_file_path can be absent in schema for some reason, though it is present in specification: https://iceberg.apache.org/spec/#manifests
+                    reference_file_path
+                        = manifest_file_deserializer.getValueFromRowByName(i, c_data_file_referenced_data_file, TypeIndex::String)
+                              .safeGet<String>();
+                }
+                catch (const DB::Exception &)
```

Comment:
Please log the errors

Reply:
`try ... catch ...` is too misleading. If we allow the field absent, why not use `if(... hasPath(c_data_file_referenced_data_file))`? Is it possible we catch other kinds of exception?